includes/wikia/services/DataMartService.class.php

```diff
@@ -201,101 +194,47 @@ public static function getTopWikisByPageviews ( $periodId, $limit = 200, Array $
+					->FROM( 'report_wiki_recent_pageviews' )
+					->AS_( 'r' )
+					->ORDER_BY( [ 'pageviews', 'desc' ] )
+					->LIMIT( $limitUsed );
+
+			if ( is_integer( $public ) ) {
+				$sql->JOIN( 'dimension_wikis' )
+					->AS_( 'd' )
 					->ON( 'r.wiki_id', 'd.wiki_id' )
-				->WHERE( 'd.public' )->EQUAL_TO( $public );
-		}
-
-		if ( !empty( $langs ) ) {
-			$sql->AND_( 'r.lang' )->IN( $langs );
-		}
-
-		if ( !empty( $hub ) ) {
-			$sql->AND_( 'r.hub_name' )->EQUAL_TO( $hub );
-		}
-
-		$topWikis = $sql->runLoop( $db, function( &$topWikis, $row ) {
-			$topWikis[$row->id] = $row->pageviews;
-		} );
-
-		$topWikis = array_slice( $topWikis, 0, $limit, true );
-
-		return $topWikis;
-	}
-
-	/**
-	 * get events by wiki Id
-	 * @param integer $periodId
-	 * @param string $startDate [YYYY-MM-DD]
-	 * @param string $endDate [YYYY-MM-DD]
-	 * @param integer $wikiId
-	 * @param string $eventType [creates/edits/deletes/undeletes]
-	 * @return array $events [ array( 'YYYY-MM-DD' => pageviews ) ]
-	 * Note: number of edits includes number of creates
-	 */
-	protected static function getEventsByWikiId ( $periodId, $startDate, $endDate = null, $wikiId = null, $eventType = null ) {
-		$app = F::app();
-
-		wfProfileIn( __METHOD__ );
+					->WHERE( 'd.public' )
+					->EQUAL_TO( $public );
+			}
 
-		if ( empty( $wikiId ) ) {
-			$wikiId = $app->wg->CityId;
-		}
+			if ( !empty( $langs ) ) {
+				$sql->AND_( 'r.lang' )->IN( $langs );
+			}
 
-		if ( empty( $endDate ) ) {
-			if ( $periodId == self::PERIOD_ID_MONTHLY ) {
-				$endDate = date( 'Y-m-01' );
-			} else {
-				$endDate = date( 'Y-m-d', strtotime( '-1 day' ) );
+			if ( !empty( $hub ) ) {
+				$sql->AND_( 'r.hub_name' )->EQUAL_TO( $hub );
 			}
-		}
 
-		$db = DataMartService::getDB();
-		$events = ( new WikiaSQL() )->skipIf( self::isDisabled() )
-			->cacheGlobal( self::TTL )
-			->SELECT( "date_format(time_id,'%Y-%m-%d')" )->AS_( 'date' )
-				->SUM( 'creates' )->AS_( 'creates' )
-				->SUM( 'edits' )->AS_( 'edits' )
-				->SUM( 'deletes' )->AS_( 'deletes' )
-				->SUM( 'undeletes' )->AS_( 'undeletes' )
-			->FROM( 'rollup_wiki_namespace_user_events' )
-			->WHERE( 'period_id' )->EQUAL_TO( $periodId )
-				->AND_( 'wiki_id' )->EQUAL_TO( $wikiId )
-				->AND_( 'time_id' )->BETWEEN( $startDate, $endDate )
-			->GROUP_BY( 'date', 'wiki_id' )
-			->runLoop( $db, function( &$events, $row ) {
-				$events[$row->date] = array(
-					'creates' => $row->creates,
-					'edits' => $row->creates + $row->edits,
-					'deletes' => $row->deletes,
-					'undeletes' => $row->undeletes,
-				);
+			$topWikis = $sql->runLoop( $db, function ( &$topWikis, $row ) {
+				$topWikis[$row->id] = $row->pageviews;
 			} );
 
-		// get data depending on eventType
-		if ( !empty( $eventType ) ) {
-			$temp = array();
-			foreach ( $events as $date => $value ) {
-				$temp[$date] = $value[$eventType];
-			}
-			$events = $temp;
-		}
-
-		wfProfileOut( __METHOD__ );
+			$topWikis = array_slice( $topWikis, 0, $limit, true );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$topWikis". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)